06_puppy/n0npax/syncStore.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"errors"
+)
+
+func NewSyncStore() *SyncStore {
+	return &SyncStore{}
+}
+
+func (m *SyncStore) CreatePuppy(puppy *Puppy) int {
```

Comment:
Here you should be making use of the `sync.Mutex` capability to put a mutex lock on the resource to ensure race conditions can't occur. If you extend your definition of the `SyncStore` you will be able to do this.

```go
type SyncStore struct {
    sync.Map
    sync.Mutex
}

func (m *SyncStore) CreatePuppy(puppy *Puppy) int {
    m.Lock()
    defer m.Unlock()

    // Continue with functionality below
}

```

Reply:
`create`, `update` & `delete` are using mutex now. Thanks!